packages/yourloops/components/patient-list/patient-list.hook.tsx

```diff
@@ -0,0 +1,293 @@
+import { PatientListColumns, PatientListTabs, PatientListFilters } from './enums/patient-list.enum'
+import { usePatientContext } from '../../lib/patient/patient.provider'
+import { useUserName } from '../../lib/custom-hooks/user-name.hook'
+import { getMedicalValues } from '../patient/utils'
+import { ActionsCell, AlarmPercentageCell, FlagIconCell, MessageCell, PendingIconCell } from './custom-cells'
+import { type Alarms } from '../../lib/patient/models/alarms.model'
+import { useAuth } from '../../lib/auth'
+import { type Patient } from '../../lib/patient/models/patient.model'
+import PatientUtils from '../../lib/patient/patient.util'
+import { type GridRowModel } from './models/grid-row.model'
+import { type UserToRemove } from '../dialogs/remove-direct-share-dialog'
+import { getPatientFullName } from 'dumb/dist/src/utils/patient/patient.util'
+import { useNavigate } from 'react-router-dom'
+import Box from '@mui/material/Box'
+import { type PatientProfile } from '../../lib/patient/models/patient-profile.model'
+import { useSortComparatorsHook } from '../../lib/custom-hooks/sort-comparators.hook'
+import { useQueryParams } from '../../lib/custom-hooks/query-params.hook'
+
+interface PatientListHookReturns {
+  columns: GridColDef[]
+  columnsVisibility: GridColumnVisibilityModel
+  selectedTab: PatientListTabs
+  inputSearch: string
+  gridApiRef: MutableRefObject<GridApiCommon>
+  paginationModel: GridPaginationModel
+  patientToRemoveForHcp: Patient | null
+  patientToRemoveForCaregiver: UserToRemove | null
+  rows: GridRowsProp
+  setInputSearch: (value: string) => void
+  setColumnsVisibility: (model: GridColumnVisibilityModel) => void
+  setPaginationModel: (model: GridPaginationModel) => void
+  onChangingTab: (newTab: PatientListTabs) => void
+  onCloseRemoveDialog: () => void
+  onRowClick: (params: GridRowParams) => void
+  toggleColumnVisibility: (columnName: PatientListColumns) => void
+}
+
+export const usePatientListHook = (): PatientListHookReturns => {
+  const { t } = useTranslation()
+  const { classes } = usePatientListStyles()
+  const { getFlagPatients, user } = useAuth()
+  const { getPatientById, filterPatients } = usePatientContext()
+  const { getUserName } = useUserName()
+  const { sortByUserName } = useSortComparatorsHook()
+  const navigate = useNavigate()
+  const gridApiRef = useGridApiRef()
+  const queryParams = useQueryParams()
+  const trNA = t('N/A')
+
+  const [selectedTab, setSelectedTab] = useState<PatientListTabs>(PatientListTabs.Current)
+  const [selectedFilter, setSelectedFilter] = useState<PatientListFilters>(PatientListFilters.All)
+  const [inputSearch, setInputSearch] = useState<string>('')
+  const [patientToRemoveForHcp, setPatientToRemoveForHcp] = useState<Patient | null>(null)
+  const [patientToRemoveForCaregiver, setPatientToRemoveForCaregiver] = useState<UserToRemove | null>(null)
+  const [paginationModel, setPaginationModel] = useState<GridPaginationModel>({ pageSize: 10, page: 0 })
```

Comment:
I don't think that this state should be here. It could be defined in the component. It is not used in the custom hook. :thinking: 

Same remark for `columnsVisibility`